client/client-lib/src/clients/gateway.rs

```diff
@@ -361,6 +374,102 @@ impl GatewayClient {
         self.context.db.apply_batch(batch).expect("DB error");
         Ok(())
     }
+
+    // FIXME: copy-pasted from UserClient
```

Comment:
What if we just gave the `GatewayClient` an instance of `UserClient`?  That's what we do in the integration tests.

Reply:
@justinmoon is fixing this rn (in another PR afaik). The idea is to have three type of configs:

```rust
struct ClientConfig {
…
}

struct UserClientConfig {
    common: ClientConfig,
    …
}

struct GatewayClientConfig {
    common: ClientConfig,
    …
}

impl AsRef<ClientConfig> for UserClientConfig {…}
impl AsRef<ClientConfig> for GatewayClientConfig {…}
```

With that we can have a client generic over a config with three implementations, a common and two specialized ones:

```rust
struct Client<T> {
    config: T,
    …
}

impl<T: AsRef<ClientConfig>> for Client<T> {
    // Common functionality
}

impl Client<UserClientConfig> {
    // User specific functionality
}

impl Client<GatewayClientConfig> {
    // Gateway specific functionality
}
```

Reply:
Fixed by #260